### PR TITLE
Adjusted lockSettingsProps for the HTML5 client

### PR DESCRIPTION
**Pull request by pedrobmarin**
Hey @ritzalam 

I did some of the work needed to all your changes to work with the HTML5 client. Couldn't figure it out about the breakout options yet. Will need some of your help on that one.

I have some suggestions about this changes:
 - could bbb-web always respond the default lock settings props? This way we would not need to store the default props in both of the clients (Flash and HTML5).
 - There are some name conventions that made this adaptations a little confuse. For instance, in the HTML5 client it's used `disablePrivChat` and `disablePubChat`. It follows the same pattern that already existed for the (I belive) the lock settings update message. Your implementation uses `disablePrivateChat` and `disablePublicChat`.
 - Still about naming, thare are some places we use `prop` and other `props`. The HTML5 client already was using `lockSettingsProp`, I had to change to `lockSettingsProps`.
 - Is the breakout options passed on the `enter` API call? Could we pass it on meeting creation? It would work better on Meteor.

### CODE REVIEW

**Comment by ritzalam** [comment]
Hi @pedrobmarin ,

Here are my responses: 
 - could bbb-web always respond the default lock settings props? This way we would not need to store the default props in both of the clients (Flash and HTML5).

We had to keep the default props on the config.xml for backward compatibility. If there are not params passed on the CREATE API call, we use what are defined in config.xml. If there is AT LEAST one param passed, then we take that param and use the defaults in bbb.properties. The idea is to deprecate the one in config.xml later on and use the ones defined server side.

 - There are some name conventions that made this adaptations a little confuse. For instance, in the HTML5 client it's used disablePrivChat and disablePubChat. It follows the same pattern that already existed for the (I belive) the lock settings update message. Your implementation uses disablePrivateChat and disablePublicChat.

 - Still about naming, thare are some places we use prop and other props. The HTML5 client already was using lockSettingsProp, I had to change to lockSettingsProps.

I did not realize the html5 client would be affected by changing the name much. I thought that the lockSettings were new params to the CreateMessage body and that the names would not affect the current html5 code. We can change it to match the html5 code if it causes issues.

 - Is the breakout options passed on the enter API call? Could we pass it on meeting creation? It would work better on Meteor.

The breakout options is passed to the client on the ENTER API call. I had it also passed on the CreateMeetingMsg but removed it. I can add it back if that would make it easier.

Please let me know.

File: bigbluebutton-html5/imports/api/meetings/server/modifiers/addMeeting.js [hunk 1]
```diff
@@ -64,6 +91,7 @@ export default function addMeeting(meeting) {
       screenshareConf: String,
     },
     metadataProp: Object,
+    lockSettingsProps: Match.Any,
```
Comment (antobinary) [inline bigbluebutton-html5/imports/api/meetings/server/modifiers/addMeeting.js]: ` Match.ObjectIncluding(`...

File: bigbluebutton-html5/imports/api/meetings/server/modifiers/addMeeting.js [hunk 2]
```diff
@@ -3,6 +3,33 @@ import { check, Match } from 'meteor/check';
 import Meetings from '/imports/api/meetings';
 import Logger from '/imports/startup/server/logger';
 
+const getLockSettings = props => {
+  if (props) {
+    // Since we miss a name convention here, we need to translate some properties
+    // key name to fit the rest of the messaging system
+    props = Object.assign(props, {
+      disablePrivChat: props.disablePrivateChat,
```
Comment (antobinary) [inline bigbluebutton-html5/imports/api/meetings/server/modifiers/addMeeting.js]: If I understand this correctly, we could adapt the html5 name to match the convention

File: bigbluebutton-html5/imports/api/meetings/server/modifiers/addMeeting.js (hunk 1 above)
Comment (pedrobmarin) [reply bigbluebutton-html5/imports/api/meetings/server/modifiers/addMeeting.js]: This value can be null, can we use this option in this case? I tried Match.Maybe(... but could not handle that to work.

File: bigbluebutton-html5/imports/api/meetings/server/modifiers/addMeeting.js (hunk 2 above)
Comment (pedrobmarin) [reply bigbluebutton-html5/imports/api/meetings/server/modifiers/addMeeting.js]: Yes, we could. But we have to take a special care about things like this:
https://github.com/bigbluebutton/bigbluebutton/blob/3775d1ebc2c3334af350562f9c9288dd79ad9b13/bigbluebutton-html5/imports/api/meetings/server/methods/toggleLockSettings.js#L25-L34

**Comment by pedrobmarin** [comment]
Hi @ritzalam. If you could add back the breakout options on the CreateMeetingMsg would be good enough. All the other things I can resolve with @antobinary

**Comment by ritzalam** [comment]
@pedrobmarin I have updated so that breakout and lock settings are passed using the create meeting message.

**Comment by pedrobmarin** [comment]
Nice. Thanks @ritzalam

**Pull request merged**